cli/cmd/deploy/deploy.go

```diff
@@ -276,6 +236,26 @@ func projectParamPrompt(ctx context.Context, c *client.Client, orgName, githubUR
 	return opts, nil
 }
 
+func credentialsPrompt(projectPath string) map[string]string {
+	csMap, err := inferSourceConnectors(projectPath)
+	if err != nil {
+		exitWithFailure(err)
+	}
+
+	creds := make(map[string]string)
+	for connector, sources := range csMap {
+		if promptFunc, ok := connectorPrompts[connector.Type]; ok {
+			fmt.Printf("\nsources %s with connector %s require credentials. Leave blank if public access enabled.\n", sources, connector.Type)
```

Comment:
The decision to leave blank for public access should IMO be part of the connector-specific prompt. Some connectors might not support a mode without credentials.